lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -1191,6 +1198,56 @@ func networkFromContainerInfo(vc *viccontainer.VicContainer, info *models.Contai
 		Networks: make(map[string]*dnetwork.EndpointSettings),
 	}
 
+	shortCID := vc.ContainerID[0:ShortIDLen]
+
+	// Fill in as much info from the endpoint struct inside of the ContainerInfo.
+	// The rest of the data must be obtained from the Scopes portlayer.
+	for _, ep := range info.Endpoints {
+		netEp := &dnetwork.EndpointSettings{
+			IPAMConfig:          nil, //Get from Scope PL
+			Links:               nil,
+			Aliases:             nil,
+			NetworkID:           "", //Get from Scope PL
+			EndpointID:          ep.ID,
+			Gateway:             ep.Gateway,
+			IPAddress:           "",
+			IPPrefixLen:         0,  //Get from Scope PL
+			IPv6Gateway:         "", //Get from Scope PL
+			GlobalIPv6Address:   "", //Get from Scope PL
+			GlobalIPv6PrefixLen: 0,  //Get from Scope PL
+			MacAddress:          "", //Container endpoints currently do not have mac addr yet
+		}
+
+		// This data may not be valid if DHCP is used.  Get from Scope PL in that case.
+		if ep.Address != "" {
+			addrs := strings.Split(ep.Address, "/")
+			netEp.IPAddress = addrs[0]
+
+			if len(addrs) > 1 {
+				if prefixLen, err := strconv.Atoi(addrs[1]); err == nil {
+					netEp.IPPrefixLen = prefixLen
+				}
+			}
+		}
+
+		if len(ep.Aliases) > 0 {
+			netEp.Aliases = make([]string, 0)
+			found := false
+			for _, alias := range ep.Aliases {
+				netEp.Aliases = append(netEp.Aliases, alias)
+				if alias == shortCID {
+					found = true
```

Comment:
Another trick could be you could just assign a nil instead of creating a new array. new array would be created as soon as first element is added.